sdk/endpoint/switch.go

```diff
@@ -0,0 +1,61 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools/spanhelper"
+)
+
+// ConnectionEndpointMatcher is a function that accepts a connection and returns
+// the corresponding network service server.
+type ConnectionEndpointMatcher func(*connection.Connection) (networkservice.NetworkServiceServer, error)
+
+// SwitchEndpoint is a NetworkServiceServer composition primitive which passes
+// incoming requests into one of its items specified by user-provided matcher
+// function.
+type SwitchEndpoint struct {
+	// endpointMap    map[string][]SwitchEndpointItem
+	items   []networkservice.NetworkServiceServer
+	matcher ConnectionEndpointMatcher
+}
+
+// NewSwitchEndpoint returns a new SwitchEndpoint.
+func NewSwitchEndpoint(items []networkservice.NetworkServiceServer, matcher ConnectionEndpointMatcher) *SwitchEndpoint {
```

Comment:
I would be suggesting to simplify API to something like:

```suggestion
func NewSwitchEndpoint(items ...networkservice.NetworkServiceServer *SwitchEndpoint {
return func NewSwitchEndpointWithCustomMatchers(items..., DefaultMatcher1(), DefaultMatcher2(), ...)
```